pkg/intercp/catalog/catalog.go

```diff
@@ -41,6 +42,19 @@ func Leader(ctx context.Context, catalog Catalog) (Instance, error) {
 	return Instance{}, ErrNoLeader
 }
 
+func InstanceOfID(ctx context.Context, catalog Catalog, id string) (Instance, error) {
+	instances, err := catalog.Instances(ctx)
+	if err != nil {
+		return Instance{}, err
+	}
+	for _, instance := range instances {
+		if instance.Id == id {
+			return instance, nil
+		}
+	}
+	return Instance{}, ErrInstanceNotFound
```

Comment:
We can't because the signature says `(Instance, error)`. I could change it to `(*Instance, error)`, but I'd try to avoid pointers to avoid nil dereference.